cmd/server/http/http.go

```diff
@@ -739,23 +698,34 @@ func convertTimeToEpoch(t time.Time) int64 {
 }
 
 type commitInfo struct {
+	ArtifactID  string
 	AuthorName  string
 	AuthorEmail string
 	Service     string
 }
 
+var extractInfoFromCommitRegex = regexp.MustCompile(`^\[(?P<service>.*)\]( artifact (?P<artifactID>[^ ]+) by)?.*\nArtifact-created-by:\s(?P<authorName>.*)\s<(?P<authorEmail>.*)>`)
+var extractInfoFromCommitRegexNamesLookup = make(map[string]int)
+
+func init() {
+	for index, name := range extractInfoFromCommitRegex.SubexpNames() {
+		if name != "" {
+			extractInfoFromCommitRegexNamesLookup[name] = index
+		}
+	}
+}
+
```

Comment:
This is clever 👏 
Could we move it into the scope of `extractInfoFromCommit` where the `pattern` was defined before? Just to avoid global scoped logic?

Reply:
missed a few first time around, including this. Yeah will look at that

Reply:
Oh, I see what you mean now. Yeah, we could do that, but this will run it as early as possible, this failing faster. Also it will only compile the regex once :)

Reply:
So will it in the handler setup. It is at startup where the server mux is created. 
Whether the error happens on importing `cmd/server/http` or using the handler function that requires the regex, I'd prefer the latter. But your call :)

Reply:
True 💪

Reply:
Done 👍